gulpfile.babel.js

```diff
@@ -0,0 +1,264 @@
+import gulp from 'gulp';
+import less from 'gulp-less';
+import cleanCss from 'gulp-clean-css';
+import sourcemaps from 'gulp-sourcemaps';
+import Autoprefix from 'less-plugin-autoprefix';
+import eslint from 'gulp-eslint';
+import imagemin from 'gulp-imagemin';
+import rollup from 'gulp-rollup';
+import uglify from 'gulp-uglify';
+import babel from 'gulp-babel';
+import rename from 'gulp-rename';
+import del from 'del';
+import nsp from 'gulp-nsp';
+import * as path from 'path';
+import changed from 'gulp-changed';
+import concat from 'gulp-concat';
+import browserSync from 'browser-sync';
+import browserify from 'browserify';
+import browserifyInc from 'browserify-incremental';
+import source from 'vinyl-source-stream';
+import buffer from 'vinyl-buffer';
+import notify from 'gulp-notify';
+import plumber from 'gulp-plumber';
+import processhtml from 'gulp-processhtml';
+import uncss from 'gulp-uncss';
+
+const isProdBuild = () => process.argv.filter(val => val.toLowerCase().indexOf('-prod') !== -1).length > 0;
+const pkgJson = require('./package.json');
+const server = browserSync.create();
+const mainDirectories = {
+	dev: './server/',
+	dist: './dist/'
+};
+
+const settings = {
+	sources: {
+		markup: './src/*.html',
+		styles: './src/assets/less/**/*.less',
+		stylesEntryPoint: './src/assets/less/index.less',
+		scripts: './src/app/**/*.js',
+		images: './src/assets/img/**/*.{png,jpg,gif,svg}',
+		externalCss: pkgJson.bootstrapKickstart.bundleCSS,
+		externalJs: pkgJson.bootstrapKickstart.bundleExternalJS,
+		staticFiles: pkgJson.bootstrapKickstart.includeStaticFiles
+	},
+	destinations: {
+		dev: {
+			markup: `${mainDirectories.dev}`,
+			styles: `${mainDirectories.dev}assets/css/`,
+			scripts: `${mainDirectories.dev}app/`,
+			images: `${mainDirectories.dev}assets/img/`,
+			libs: `${mainDirectories.dev}libs/`
+		},
+		prod: {
+			markup: `${mainDirectories.dist}`,
+			styles: `${mainDirectories.dist}assets/css/`,
+			scripts: `${mainDirectories.dist}app/`,
+			images: `${mainDirectories.dist}assets/img/`,
+			libs: `${mainDirectories.dist}libs/`
+		}
+	},
+	autoPrefix: [
+		'> 1%',
+		'last 3 version',
+		'ie 8',
+		'ie 9',
+		'Firefox ESR',
+		'Opera 12.1'
+	]
+};
+
+function onError(err) {
+	notify({
+		title: 'Gulp Task Error',
+		subtitle: 'Plugin: <%= error.plugin %>',
+		message: 'Check the console.'
+	}).write(err);
+
+	console.log(err.toString());
+
+	this.emit('end');
+}
+
+export function clean() {
```

Comment:
Deleting lots of files on every (re)start destroys SSDs a lot faster than on normal use. Let's talk about the pros and cons.